File: dnas/sensemaker/zomes/integrity/sensemaker/src/assessment.rs
use crate::method::DataSet;
use crate::range::RangeValue;
use hdi::prelude::*;

#[hdk_entry_helper]
#[derive(Clone)]
pub struct Assessment {
    pub value: RangeValue,
    pub dimension_eh: EntryHash,
    pub resource_eh: EntryHash,
    pub resource_type_eh: EntryHash,
    pub maybe_input_dataset: Option<DataSet>,
    pub author: AgentPubKey,
}

File: types/src/assessment.ts
import { AgentPubKey, EntryHash } from "@holochain/client"
import { RangeValue } from "./range"
import { DataSet } from "./method"
import { Option } from "./utils"

export interface CreateAssessmentInput {
    value: RangeValue,
    dimension_eh: EntryHash,
    subject_eh: EntryHash,
    maybe_input_dataSet: Option<DataSet>, // For objective Dimensions only
}

export type Assessment = CreateAssessmentInput & {
    author: AgentPubKey
}

export interface GetAssessmentsForResourceInput {
    resource_eh: EntryHash,
    dimension_eh: EntryHash,
}
